pkg/resource/v1/certconfig/delete.go

```diff
@@ -68,3 +93,13 @@ func (r *Resource) newDeleteChangeForUpdatePatch(ctx context.Context, obj, curre
 
 	return certConfigsToDelete, nil
 }
+
+func newDeleteOptions() *apismetav1.DeleteOptions {
+	propagation := apismetav1.DeletePropagationForeground
+
+	options := &apismetav1.DeleteOptions{
+		PropagationPolicy: &propagation,
```

Comment:
Why do we need to configure this? I only know we need this for deployments in KVM but the cert configs should be rather standalone.

Reply:
No specific reason. `kvm-operator` is probably where I noticed this and thought that it wouldn't hurt to have it by default. When there's nothing to propagate to it's almost no-op (I'd imagine it API still goes through the stages to update `deletionTimestamp` etc.) and when there is something, it takes care of cleaning it properly. I have no strong opinions on this. If you think this should be only used when necessary, I'm ok with that. WDYT?

Reply:
Uncertain. We should use this when it is necessary and know what we are doing and why. Lets maybe better go with the default.